src/tests/unit/process-get-filters.test.js

```diff
@@ -0,0 +1,133 @@
+import chai from 'chai';
+
+import { GetFilterProcessor } from 'utils/process-get-filters';
+
+import { processGetFiltersData } from './test-data';
+
+chai.should();
+
+describe('normalizeFilterNames', () => {
+  const {
+    fooFilter, normalizedFooFilter,
+    nameFilter, normalizedNameFilter,
+  } = processGetFiltersData;
+  describe('when it gets an empty object', () => {
+    it('returns an empty object', () => {
+      const normalizedFilters = GetFilterProcessor.normalizeFilterNames({});
+      normalizedFilters.should.deep.equal({});
+    });
+  });
+  describe('when it gets a non empty object', () => {
+    it('changes filter keys but leaves non-filter keys unchanged', () => {
+      const normalizedFilters = GetFilterProcessor.normalizeFilterNames({
+        ...fooFilter,
+        ...nameFilter,
+      });
+      normalizedFilters.should.deep.equal({
+        ...normalizedFooFilter,
+        ...normalizedNameFilter,
+      });
+    });
+  });
+});
+
+describe('normalizeFilterName', () => {
+  describe('when it gets a filter string', () => {
+    it('returns only the text inside the brackets', () => {
+      const normalizedFilter = GetFilterProcessor.normalizeFilterName('filter[foo]');
+      normalizedFilter.should.equal('foo');
+    });
+    describe('when the normalized name is `filter`', () => {
+      it('still returns only the name inside the brackets', () => {
+        const normalizedFilter = GetFilterProcessor.normalizeFilterName('filter[filter]');
+        normalizedFilter.should.equal('filter');
+      });
+    });
+  });
+  describe('when it gets a non filter string', () => {
+    it('returns the string verbatim', () => {
+      const normalizedFilter = GetFilterProcessor.normalizeFilterName('foo');
+      normalizedFilter.should.equal('foo');
+    });
+    describe('when the string is `filter`', () => {
+      it('still returns the name verbatim', () => {
+        const normalizedFilter = GetFilterProcessor.normalizeFilterName('filter');
+        normalizedFilter.should.equal('filter');
+      });
+    });
+  });
+});
+
+describe('processGetFilters', () => {
+  let processor;
+  const {
+    fooParamName,
+    nameParamName,
+    fooColumnName,
+    nameColumnName,
+    fooFilter,
+    fooBindParams,
+    fooConditional,
+    nameFilter,
+    nameBindParams,
+    nameConditional,
+    bahFilter,
+    emptyBindParams,
+    emptyConditional,
+  } = processGetFiltersData;
+  beforeEach(() => {
+    processor = new GetFilterProcessor(
+      [
+        fooParamName, nameParamName,
```

Comment:
Could you put a newline between these two since the array is split over multiple lines?

Reply:
might move these to just all be on one line instead since it's a short array